app/src/main/java/de/danoeh/antennapod/discovery/ItunesTopListLoader.java

```diff
@@ -23,23 +24,33 @@
 import java.util.Locale;
 import java.util.concurrent.TimeUnit;
 
+import static android.content.Context.MODE_PRIVATE;
+
 public class ItunesTopListLoader {
     private static final String TAG = "ITunesTopListLoader";
     private final Context context;
+    public static final String PREF_KEY_COUNTRY_CODE = "country_code";
+    public static final String PREFS = "CountryRegionPrefs";
 
     public ItunesTopListLoader(Context context) {
         this.context = context;
     }
 
-    public Single<List<PodcastSearchResult>> loadToplist(int limit) {
+    public Single<List<PodcastSearchResult>> loadToplist() {
+        String defaultCountry = Locale.getDefault().getCountry();
+        SharedPreferences prefs = context.getSharedPreferences(PREFS, MODE_PRIVATE);
+        String countryCode = prefs.getString(PREF_KEY_COUNTRY_CODE, defaultCountry);
+        return this.loadToplist(countryCode, 25);
+    }
+
+    public Single<List<PodcastSearchResult>> loadToplist(String country, int limit) {
         return Single.create((SingleOnSubscribe<List<PodcastSearchResult>>) emitter -> {
-            String country = Locale.getDefault().getCountry();
             OkHttpClient client = AntennapodHttpClient.getHttpClient();
             String feedString;
             try {
                 feedString = getTopListFeed(client, country, limit);
             } catch (IOException e) {
-                feedString = getTopListFeed(client, "us", limit);
+                feedString = getTopListFeed(client, "US", limit);
```

Comment:
What do you think about only falling back to `US` when no country was selected yet? Other countries could display something like "Sorry, we do not have suggestions for your region".

Reply:
up in line 40, I pick the default country of the device, so there will always be a country even if you don't select it.  I personally don't wan to default to `US` all the time, but that is the safe option.

Reply:
Displaying the US as fallback makes it harder to see if a region supports localized suggestions. If I explicitly select a region, I would expect to either see suggestions from that region or an error message, not US suggestions. Contrarily, if I use the app without ever selecting a region, I would expect to get suggestions, even if they are not completely relevant.

What about something like this?

```
String countryCode = prefs.getString(PREF_KEY_COUNTRY_CODE, COUNTRY_CODE_UNSET);
return this.loadToplist(countryCode, 25);
```

```
String loadCountry = country;
if (COUNTRY_CODE_UNSET.equals(country)) {
    loadCountry = Locale.getDefault().getCountry();
}
try {
    feedString = getTopListFeed(client, country, limit);
} catch (IOException e) {
    if (COUNTRY_CODE_UNSET.equals(country)) {
        feedString = getTopListFeed(client, "US", limit);
    }
}
```

Reply:
Great idea. Togo doesn't have data.  The new code tells the user

![Screenshot_20201011-163744](https://user-images.githubusercontent.com/149837/95693283-692bf380-0be0-11eb-8026-9286e6aa05a1.jpg)